bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/AuthenticationApi.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.netatmo.internal.api;
+
+import static org.openhab.core.auth.oauth2client.internal.Keyword.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.binding.netatmo.internal.api.data.NetatmoConstants.FeatureArea;
+import org.openhab.binding.netatmo.internal.api.dto.NAAccessTokenResponse;
+import org.openhab.binding.netatmo.internal.config.NetatmoBindingConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Allows access to the AutomowerConnectApi
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+class AuthenticationApi extends RestManager {
+    private final Logger logger = LoggerFactory.getLogger(AuthenticationApi.class);
+    private final NetatmoBindingConfiguration configuration;
+    private final ScheduledExecutorService scheduler;
+    private @Nullable ScheduledFuture<?> refreshTokenJob;
+
+    AuthenticationApi(ApiBridge apiBridge, NetatmoBindingConfiguration configuration,
+            ScheduledExecutorService scheduler) {
+        super(apiBridge, FeatureArea.NONE);
+        this.configuration = configuration;
+        this.scheduler = scheduler;
+    }
+
+    void authenticate() throws NetatmoException {
+        Map<String, @Nullable String> payload = new HashMap<>(Map.of(SCOPE, FeatureArea.ALL_SCOPES));
```

Comment:
is that not enough/valid ?
`         Map<String, @Nullable String> payload = Map.of(SCOPE, FeatureArea.ALL_SCOPES);
`

Reply:
No because this would produce an immutable Map and I need to add some elements later on.

Reply:
Ok